eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryImpl.java

```diff
@@ -113,27 +120,49 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantEn
 
     }
 
-    private Long countByComplexFiltersHelper(Query query, List<VariantEntityRepositoryFilter> filters) {
-        applyFilters(query, filters);
+    private Long countByComplexFiltersHelper(Criteria existingCriteria, List<VariantEntityRepositoryFilter> filters) {
+        List<Criteria> criteriaList = getFiltersCriteria(filters);
+        criteriaList.add(existingCriteria);
+        Criteria criteria = new Criteria().andOperator(criteriaList.toArray(new Criteria[criteriaList.size()]));
+
+        Aggregation aggregation = Aggregation.newAggregation(
+                Aggregation.match(criteria),
+                Aggregation.group().count().as("count")
+        );
+
+        AggregationResults<VariantAggregationCount> aggregationResults =
+                mongoTemplate.aggregate(aggregation, VariantEntity.class, VariantAggregationCount.class);
 
-        return mongoTemplate.count(query, VariantEntity.class);
+        return aggregationResults.getMappedResults().size() > 0
+                ? aggregationResults.getMappedResults().get(0).getCount() : 0;
```

Comment:
getRawResults doesn't make the value easy to access, so instead of accessing the count like:

`aggregationResults.getMappedResults().get(0).getCount()`

it'd need to be accessed like:

`(Long) ((BasicDBObject) ((BasicDBList) aggregationResults.getRawResults().get("result")).get(0)).get("count")`